### PR TITLE
podman: selinux variant, init script for background service, updated …

### PR DESCRIPTION
**Pull request by oskarirauta**
Maintainer: Daniel Golle / @dangowrt
Compile tested: x86_64, xeon powered server, OpenWRT snapshot (recent)
Run tested: x86_64, xeon powered server, OpenWRT snapshot (recent)

Description:
 - Added selinux support as a variant
 - Init script for socket
 - Updated containers.conf
 - preselected overlay as storage driver
 - changed maintainer from Daniel to me as per request.

Source code has remained same and no patches provided this time.

### CODE REVIEW

File: utils/podman/Makefile [hunk 1]
```diff
@@ -38,22 +36,54 @@ define Download/default-policy
   HASH:=cddfaa8e6a7e5497b67cc0dd8e8517058d0c97de91bf46fff867528415f2d946
 endef
 
-define Package/podman
+define Package/podman/Default
   SECTION:=utils
   CATEGORY:=Utilities
   TITLE:=Podman
   URL:=https://podman.io
-  DEPENDS:=$(GO_ARCH_DEPENDS) +conmon +cni +cni-plugins +btrfs-progs +glib2 +gnupg2 +iptables +libgpg-error +libseccomp +libgpgme +nsenter +zoneinfo-simple
```
Comment (dangowrt) [inline utils/podman/Makefile:46]: Please leave common dependencies in the `Package/podman/Default` section.

File: utils/podman/Makefile [hunk 2]
```diff
@@ -38,22 +36,54 @@ define Download/default-policy
...
+endef
+
+define Package/podman
+  $(call Package/podman/Default)
+  DEPENDS:=$(GO_ARCH_DEPENDS) +conmon +cni +cni-plugins +btrfs-progs +glib2 +gnupg2 +iptables +libgpg-error +libseccomp +libgpgme +nsenter +zoneinfo-simple +kmod-veth
```
Comment (dangowrt) [inline utils/podman/Makefile]: ...and remove them from here.

File: utils/podman/Makefile [hunk 3]
```diff
@@ -38,22 +36,54 @@ define Download/default-policy
...
+  CONFLICTS:=podman-selinux
+  VARIANT:=default
+  PROVIDES:=podman
+endef
+
+define Package/podman-selinux
+  $(call Package/podman/Default)
+  TITLE += with SELinux support
+  DEPENDS:=$(GO_ARCH_DEPENDS) +conmon +cni +cni-plugins +btrfs-progs +glib2 +gnupg2 +iptables +libgpg-error +libseccomp +libgpgme +nsenter +zoneinfo-simple +kmod-veth +libselinux
```
Comment (dangowrt) [inline utils/podman/Makefile]: use
```
DEPENDS+= libselinux
```
instead.

File: utils/podman/files/containers.conf [hunk 4]
```diff
@@ -7,19 +7,19 @@ network_config_dir="/etc/cni/net.d/"
 default_network="podman"
 
 [engine]
-runtime="/usr/sbin/crun"
+runtime="/usr/bin/crun"
```
Comment (dangowrt) [inline utils/podman/files/containers.conf:10]: Good catch.

File: utils/podman/Makefile [hunk 5]
```diff
@@ -54,7 +54,7 @@ endef
 define Package/podman-selinux
   $(call Package/podman/Default)
   TITLE += with SELinux support
-  DEPENDS:=$(GO_ARCH_DEPENDS) +conmon +cni +cni-plugins +btrfs-progs +glib2 +gnupg2 +iptables +libgpg-error +libseccomp +libgpgme +nsenter +zoneinfo-simple +kmod-veth +libselinux
+  DEPENDS:=@!arc $(GO_ARCH_DEPENDS) +conmon +cni +cni-plugins +btrfs-progs +glib2 +gnupg2 +iptables +libgpg-error +libseccomp +libgpgme +nsenter +zoneinfo-simple +kmod-veth +libselinux
```
Comment (dangowrt) [inline utils/podman/Makefile:57]: This should go to libseccomp rather than being a dependency of podman itself.

**Review by dangowrt — CHANGES_REQUESTED** [review]
Please squash everything into one commit (I guess the 2nd commit isn't needed any more actually, and if so, we should fix that in libseccomp).
The title of the commit is also too long (60 chars max), please have a more general title and also add commit description, like
```
podman: several improvements and add selinux variant

Add selinux variant, init script for background service, updated
configurations, preselected storage driver, maintainer change as
requested.
```

**Comment by dangowrt** [comment]
Please use this way to **replace** the existing commits in the PR branch rather than amending additional commits.
```
git rebase -i origin/master
# now use `fixup` or `squash` to combine them, remove unneeded lines
git push -f git@github.com:oskarirauta/packages.git HEAD:crun-pkg
```

**Comment by oskarirauta** [comment]
Replaced by pull request #15312

**Comment by oskarirauta** [comment]
Replaced by #15312 - closing this one now.

**Comment by dangowrt** [comment]
No need to close this and make a new PR, you can **edit** the git history using `git rebase -i` and then force-push onto the release branch.

**Comment by oskarirauta** [comment]
Yes. But this was faster and simpler, I am at work now, so I had to speed up things a bit. 

Lähetetty iPhonesta

**Pull request closed without merging**